### PR TITLE
port copy-syntax-table

### PR DESCRIPTION
**Pull request by brotzeit**
I don't know how to handle `Vstandard_syntax_table`.

### CODE REVIEW

File: rust_src/src/remacs_sys.rs [hunk 1]
```diff
@@ -75,6 +75,7 @@ pub enum BoolVectorOp {
 extern "C" {
     // these weren't declared in a header, for example
     pub static Vprocess_alist: Lisp_Object;
+    pub static Vstandard_syntax_table: Lisp_Object;
```
Comment (db48x) [inline rust_src/src/remacs_sys.rs]: This is wrong. There's a `syntax_table` field on every `buffer`, and `Vstandard_syntax_table` is a macro that refers to the `syntax_table` field on a particular `buffer` struct which was set aside for the purpose of holding the default values for this and other fields. See https://github.com/Wilfred/remacs/blob/master/src/buffer.h#L1086-L1093

Comment (shaleh) [reply rust_src/src/remacs_sys.rs]: https://github.com/Wilfred/remacs/blob/cdf7a08ef6a76a220661b9e6223d547780b825b6/src/syntax.h#L34

You could write a Rust macro to keep the idea.

File: rust_src/src/remacs_sys.rs [hunk 2]
```diff
@@ -154,6 +155,7 @@ extern "C" {
         dest: Lisp_Object,
         op: BoolVectorOp,
     ) -> Lisp_Object;
```
Comment (shaleh) [inline rust_src/src/remacs_sys.rs]: You should not need to manually add functions. Instead, make sure the function is exported in a C header.

**Comment by brotzeit** [comment]
Should be ok.

File: rust_src/src/syntax.rs [hunk 3]
```diff
@@ -65,4 +67,29 @@ pub fn standard_syntax_table() -> LispCharTableRef {
     unsafe { buffer_defaults.syntax_table_ }.into()
 }
 
+/// Construct a new syntax table and return it.
+/// It is a copy of the TABLE, which defaults to the standard syntax table.
+#[lisp_fn(min = "0")]
+pub fn copy_syntax_table(mut table: LispObject) -> LispObject {
+    let buffer_table = unsafe { buffer_defaults.syntax_table_ };
+    if table.is_not_nil() {
```
Comment (shaleh) [inline rust_src/src/syntax.rs]: Why not include the port of `check_syntax_table`?

Comment (shaleh) [reply rust_src/src/syntax.rs]: You can leave it off for now if you like.

**Pull request merged**